synapse/handlers/oidc_handler.py

```diff
@@ -869,73 +865,50 @@ async def _map_userinfo_to_user(
         # to be strings.
         remote_user_id = str(remote_user_id)
 
-        # first of all, check if we already have a mapping for this user
-        previously_registered_user_id = await self._sso_handler.get_sso_user_by_remote_user_id(
-            self._auth_provider_id, remote_user_id,
+        # Older mapping providers don't accept the `failures` argument, so we
+        # try and detect support.
+        mapper_args = inspect.getfullargspec(
+            self._user_mapping_provider.map_user_attributes
         )
-        if previously_registered_user_id:
-            return previously_registered_user_id
+        supports_failures = "failures" in mapper_args.args
```

Comment:
We should probably move to using `inspect.signature` as recommended by the docs of `inspect.getfullargspec`. (I think this is fine, though I always have to look up the behaviour)

Would look something like:

```python
sig = inspect.signature(self._user_mapping_provider.map_user_attributes)
supports_failures = "failures" in sig.parameters
```

Reply:
I had no idea this was deprecated! I'll make that change.

Reply:
Updated.

Reply:
Me neither till I went to have a look at the docs!